js/src/instructions/withdraw.ts

```diff
@@ -0,0 +1,102 @@
+import * as BufferLayout from '@solana/buffer-layout';
+import { TOKEN_PROGRAM_ID } from '@solana/spl-token';
+import {
+  PublicKey,
+  StakeProgram,
+  SystemProgram,
+  SYSVAR_CLOCK_PUBKEY,
+  TransactionInstruction,
+} from '@solana/web3.js';
+import BN from 'bn.js';
+import type { Lamports, ProgramAddresses, Snapshot } from '../types';
+import {
+  findAuthorityProgramAddress,
+  getHeaviestValidatorStakeAccount,
+} from '../utils';
+
+export const getWithdrawInstruction = async (
+  snapshot: Snapshot,
+  programAddresses: ProgramAddresses,
+  stSolAccountOwnerAddress: PublicKey,
+  senderStSolAccountAddress: PublicKey,
+  recipientStakeAccountAddress: PublicKey,
+  amount: Lamports
+) => {
+  const stakeAuthorityAddress = await findAuthorityProgramAddress(
+    programAddresses,
+    'stake_authority'
+  );
+  const heaviestValidatorStakeAccount =
+    getHeaviestValidatorStakeAccount(snapshot);
+
+  if (amount.lamports.lte(snapshot.stakeAccountRentExemptionBalance.lamports)) {
+    throw new Error('Amount must be greater than the rent exemption balance');
+  }
+
+  const maxWithdrawAmount = heaviestValidatorStakeAccount.balance.lamports
+    .div(new BN(10))
+    .add(new BN(10));
```

Comment:
The max withdraw amount is 1/10 of the balance + 10 **SOL**, not 10 Lamports! This should be `10_000_000_000`. (Does TypeScript have numeric underscores? Or is there a `LAMPORTS_PER_SOL` constant somewhere?)

Reply:
Ohhh yes, thanks for pointing this. Would have missed this